packages/edit-navigation/src/store/reducer.js

```diff
@@ -0,0 +1,66 @@
+/**
+ * WordPress dependencies
+ */
+import { combineReducers } from '@wordpress/data';
+
+export function mapping( state, { type, postId, ...rest } ) {
+	if ( type === 'SET_MENU_ITEM_TO_CLIENT_ID_MAPPING' ) {
+		return { ...state, [ postId ]: rest.mapping };
+	}
+
+	return state || {};
+}
+
+export function processingQueue( state, { type, postId, ...rest } ) {
+	switch ( type ) {
+		case 'START_PROCESSING_POST':
+			return {
+				...state,
+				[ postId ]: {
+					...state[ postId ],
+					inProgress: true,
+				},
+			};
+
+		case 'FINISH_PROCESSING_POST':
+			return {
+				...state,
+				[ postId ]: {
+					...state[ postId ],
+					inProgress: false,
+				},
+			};
+
+		case 'POP_PENDING_ACTION':
+			const postState = state[ postId ] || {};
+			if ( 'pendingActions' in postState ) {
+				postState.pendingActions = postState.pendingActions?.filter(
```

Comment:
A reducer shouldn't mutate the `state` object directly which is what this will do because `postState` may be a reference to `state[ postId ]`.

Reply:
Good spot! I missed that one.